Backpack-SwiftUI/CardCarousel/Classes/BPKCardCarousel.swift

```diff
@@ -0,0 +1,173 @@
+import SwiftUI
+
+struct BPKCardCarousel<Content: View>: View {
+    let cards: [Content]
+    let startingIndex: Int
+    let onCardChange: () -> Void
+    
+    init(
+        cards: [Content],
+        startingIndex: Int = 1,
+        onCardChange: @escaping () -> Void = { }
+    ) {
+        self.cards = cards
+        self.startingIndex = startingIndex
+        self.onCardChange = onCardChange
+    }
+    
+    var body: some View {
+        GeometryReader { reader in
+            InternalCardCarousel(
+                size: reader.size,
+                content: cards,
+                startingIndex: startingIndex,
+                onCardChange: onCardChange
+            )
+                .frame(
+                    width: reader.size.width,
+                    height: reader.size.height)
+        }
+    }
+}
+
+internal struct InternalCardCarousel<Content: View>: View {
+    private let size: CGSize
+    private let cardCount: Int
+    private let cards: [Content]
+    private let onCardChange: () -> Void
+    
+    private let dragAnimation: Animation = .snappy(duration: 0.5)
+    private let cardWidth: CGFloat
+    
+    @State private var currentCardIndex: Int
+    @State private var isDragging: Bool = false
+    @State private var totalDrag: CGFloat = 0.0
+    
+    private var offset: CGFloat {
+        let normalizedIndex = CGFloat(currentCardIndex) / CGFloat(cardCount - 1)
+        let cardWidthScale = cardWidth * (1 - normalizedIndex)
+        let totalCardWidth = CGFloat(cardCount) * cardWidth
+        let avgCardWidth = totalCardWidth / CGFloat(cardCount)
+        let currentCardOffset = normalizedIndex * avgCardWidth
+
+        // Calculate final offset
+        return cardWidthScale + totalCardWidth - size.width / CGFloat(cardCount) - currentCardOffset - 25.0
+    }
+    
+    init(
+        size: CGSize,
+        content: [Content],
+        startingIndex: Int,
+        onCardChange: @escaping () -> Void
+    ) {
+        self.size = size
+        self.cardCount = content.count
+        self.cardWidth = size.width * 0.8
+        self.cards = content + content
+        self.onCardChange = onCardChange
+        
+        currentCardIndex = cardCount + startingIndex
```

Comment:
On second thought, my above suggestion about making the currentCardIndex a Binding may not apply. as your calculations are done based on the dragging gesture and not the current index itself I believe? worth exploring though!